src/security/SSHAgent.ts

```diff
@@ -0,0 +1,219 @@
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Get an SSH key for authentication
+   * @param {string} pushId The push ID associated with the key
+   * @return {Buffer | null} The private key or null if not found/expired
+   */
+  getPrivateKey(pushId: string): Buffer | null {
+    const keyInfo = this.keyStore.get(pushId);
+    if (!keyInfo) {
+      return null;
+    }
+
+    // Check if key has expired
+    if (new Date() > keyInfo.expiry) {
+      console.warn(`[SSH Agent] SSH key for push ${pushId} has expired`);
+      this.removeKey(pushId);
+      return null;
+    }
+
+    return keyInfo.privateKey;
+  }
+
+  /**
+   * Check if a key exists for a push
+   * @param {string} pushId The push ID to check
+   * @return {boolean} True if key exists and is valid
+   */
+  hasKey(pushId: string): boolean {
+    const keyInfo = this.keyStore.get(pushId);
+    if (!keyInfo) {
+      return false;
+    }
+
+    // Check if key has expired
+    if (new Date() > keyInfo.expiry) {
+      this.removeKey(pushId);
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * List all active keys (for debugging/monitoring)
+   * @return {Array} Array of key information (without private keys)
+   */
+  listKeys(): Array<{ pushId: string; comment: string; expiry: Date }> {
+    const keys: Array<{ pushId: string; comment: string; expiry: Date }> = [];
+
+    for (const entry of Array.from(this.keyStore.entries())) {
+      const [pushId, keyInfo] = entry;
+      if (new Date() <= keyInfo.expiry) {
+        keys.push({
+          pushId,
+          comment: keyInfo.comment,
+          expiry: keyInfo.expiry,
+        });
+      } else {
+        // Clean up expired key
+        this.removeKey(pushId);
+      }
+    }
+
+    return keys;
+  }
+
+  /**
+   * Clean up all expired keys
+   * @return {number} Number of keys cleaned up
+   */
+  cleanupExpiredKeys(): number {
+    let cleanedCount = 0;
+    const now = new Date();
+
+    for (const entry of Array.from(this.keyStore.entries())) {
+      const [pushId, keyInfo] = entry;
+      if (now > keyInfo.expiry) {
+        this.removeKey(pushId);
+        cleanedCount++;
+      }
+    }
+
+    if (cleanedCount > 0) {
+      console.log(`[SSH Agent] Cleaned up ${cleanedCount} expired SSH keys`);
+    }
+
+    return cleanedCount;
+  }
+
+  /**
+   * Sign data with an SSH key (for SSH authentication challenges)
+   * @param {string} pushId The push ID associated with the key
+   * @param {Buffer} data The data to sign
+   * @return {Buffer | null} The signature or null if failed
+   */
+  signData(pushId: string, data: Buffer): Buffer | null {
+    const privateKey = this.getPrivateKey(pushId);
+    if (!privateKey) {
+      return null;
+    }
+
+    try {
+      // Create a sign object - this is a simplified version
+      // In practice, you'd need to handle different key types (RSA, Ed25519, etc.)
+      const sign = crypto.createSign('SHA256');
```

Comment:
Commenting for visibility - do we want to support different key types at this moment? If so, how can we make this a bit more flexible?